packages/@tailwindcss-upgrade/src/codemods/migrate-at-config.ts

```diff
@@ -0,0 +1,109 @@
+import path from 'node:path'
+import { AtRule, type Plugin, type Root } from 'postcss'
+import { normalizePath } from '../../../@tailwindcss-node/src/normalize-path'
+import type { Stylesheet } from '../stylesheet'
+import { walk, WalkAction } from '../utils/walk'
+
+export function migrateAtConfig(
+  sheet: Stylesheet,
+  { configFilePath }: { configFilePath: string },
+): Plugin {
+  function injectInto(sheet: Stylesheet) {
+    let root = sheet.root
+
+    // We don't have a sheet with a file path
+    if (!sheet.file) return
+
+    // Skip if there is already a `@config` directive
+    {
+      let hasConfig = false
+      root.walkAtRules('config', () => {
+        hasConfig = true
+        return false
+      })
+      if (hasConfig) return
+    }
+
+    // Figure out the path to the config file
+    let sheetPath = sheet.file
+    let configPath = configFilePath
+
+    let relative = path.relative(path.dirname(sheetPath), configPath)
+    if (relative[0] !== '.') {
+      relative = `./${relative}`
+    }
+    // Ensure relative is a posix style path since we will merge it with the
+    // glob.
+    relative = normalizePath(relative)
+
+    // Inject the `@config` in a sensible place
+    // 1. Above the first `@theme`
+    // 2. Below the last `@import`
+    // 3. At the top of the file
```

Comment:
This is going to be the place for adding the CSS config if we can migrate the JS config. Maybe we can make the "what is injected" here dynamic so we can decide in the JS config migrator what we need in the css?

Reply:
Yep I think once this PR is merged that we can abstract this to a function if we want. Not sure if that's necessary for this PR yet. 

There might be some additional logic where we might want to control the order of `@config`, `@theme`, injected `@variant`s and so on.

Reply:
Haha yep I would need it right away, let's pair on this afterwards 👍